### PR TITLE
Add an 'x' button to close the error messages box #1403

### PR DESCRIPTION
**Pull request by AnnMarieW**
This is a potential solution for https://github.com/plotly/dash/issues/1403
It works in my local environment, but needs better css for the x button

### CODE REVIEW

**Comment by alexcjohnson** [comment]
@AnnMarieW this works great! Re: styling, I'm thinking we can use a much lighter button right in the corner, something like this:
Screen Shot 2020-12-22 at 12 05 54 PM

Which I got from applying the following styling in the devtools, along with replacing the SVG icon with just a unicode `&times;` (or in fact just the unicode character ×)

```
<div style="
    position: absolute;
    right: 0;
    top: 0;
    color: #ccc;
    font-size: 20px;
    cursor: pointer;
">×</div>
```
Also to make `position: absolute;` work on this element, you have to define `position: relative;` on the `.dash-error-card__topbar` parent.

And the close button will want some `:hover` css to make it appear clickable.

**Comment by AnnMarieW** [comment]
That looks so much better, Thanks!

File: dash-renderer/src/components/error/FrontEnd/FrontEndError.css [hunk 1]
```diff
@@ -2,6 +2,22 @@
...
+    top: 0;
+    color: #ccc;
+    font-size: 20px;
+    cursor: pointer;
+    margin-right: 5px
+}
+
+.dash-fe-error__icon-x:hover
+{
+ color:black;
```
Comment (alexcjohnson) [inline dash-renderer/src/components/error/FrontEnd/FrontEndError.css:17]: That's probably a stronger effect than we want - really we just need enough to be able to see a change in the X. Closest analog I see is the grey debug menu buttons, which starts out about the same as `#ccc` (just a tiny bit tilted toward blue)

https://github.com/plotly/dash/blob/e7f4984d0b58511cdd8d4c97ac5ecf963dd89c35/dash-renderer/src/components/error/menu/DebugMenu.css#L70

And on hover it gets just a bit darker

https://github.com/plotly/dash/blob/e7f4984d0b58511cdd8d4c97ac5ecf963dd89c35/dash-renderer/src/components/error/menu/DebugMenu.css#L84

I'm not sure where those exact colors came from, but we may be able to just copy them both.

Comment (AnnMarieW) [reply dash-renderer/src/components/error/FrontEnd/FrontEndError.css:17]: I see, I'll make the change.  Thanks!

File: dash-renderer/src/components/error/FrontEnd/FrontEndErrorContainer.react.js [hunk 2]
```diff
@@ -34,6 +34,12 @@ class FrontEndErrorContainer extends Component {
                         </strong>
                         ){connected ? null : '\u00a0 🚫 Server Unavailable'}
                     </div>
+                    <div
+                        className='dash-fe-error__icon-x'
+                        onClick={() => clickHandler()}
+                    >
+                        x
```
Comment (alexcjohnson) [inline dash-renderer/src/components/error/FrontEnd/FrontEndErrorContainer.react.js:41]:
```suggestion
                        ×
```
This is the `&times;` character, which makes a nicer button than the letter x.

File: dash-renderer/src/components/error/icons/OffIcon.svg [hunk 3]
```diff
@@ -1,3 +1,3 @@
 <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 54 24">
-<path d="M18 2l18 18m0-18l-18 18" stroke-width="4" stroke="#fff"></path>
+<path d="M18 2l18 18m0-18l-18 18" stroke-width="6" stroke="#fff"></path>
```
Comment (alexcjohnson) [inline dash-renderer/src/components/error/icons/OffIcon.svg:2]:
```suggestion
<path d="M18 2l18 18m0-18l-18 18" stroke-width="4" stroke="#fff"></path>
```
Not used in this PR anymore so let's put the original back.

**Review by alexcjohnson — APPROVED** [review]
Just two minor final edits then this is ready to go! 💃

**Pull request merged**